glean-core/python/glean/glean.py

```diff
@@ -55,11 +57,18 @@ class Glean:
     # and saved between test runs.
     _ping_type_queue: Set["PingType"] = set()
 
+    # The application id to send in the ping.
+    _application_id: str = "glean-python"
```

Comment:
Let's not provide defaults for this, if possible. We should rather make the configuration object required, if needed: otherwise if anyone else starts using these bindings without paying attention to the docs, we'd end up inflating this app id. Let's be explicit!

Reply:
I'd like to think on this one.  As this library is primarily being used for writing tests, requiring a configuration object means that we can no longer provide a testing decorator to reset Glean for each test (since those don't take arguments).  We'd need a way to set it globally in a way that would survive Glean resets (or else make the API for testing far less convenient and error-prone by requiring users to reset Glean themselves).

Filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1595486)